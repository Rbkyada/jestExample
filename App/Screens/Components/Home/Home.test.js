import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import combineReducers from '../../../Reducers/index';
import {createStore} from 'redux';
import {fireEvent, render} from '@testing-library/react-native';
import Home from './Home';

jest.mock('redux-saga', () => () => ({run: jest.fn()}));
jest.mock('redux-persist', () => ({
  persistStore: jest.fn(),
  persistReducer: jest.fn(),
}));
jest.mock('react-native-config');

const UseEffect = callback => {
  React.useEffect(callback);
  return null;
};

const eventData = {
  nativeEvent: {
    contentOffset: {
      x: 0,
      y: 425,
    },
    contentSize: {
      // Dimensions of the scrollable content
      height: 885,
      width: 328,
    },
    layoutMeasurement: {
      // Dimensions of the device
      height: 469,
      width: 328,
    },
  },
};

const Store = createStore(combineReducers);

const component = (
  <Provider store={Store}>
    <Home />
  </Provider>
);

it('renders correctly', () => {
  const tree = render(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('when Flatlist is Empty', () => {
  const {getByTestId} = render(component);
  expect(getByTestId('flatlist').props.data.length).toBe(0);
});

test('render Flatlist data', () => {
  const {getByTestId} = render(component);
  fireEvent.scroll(getByTestId('flatlist'), eventData);
});
