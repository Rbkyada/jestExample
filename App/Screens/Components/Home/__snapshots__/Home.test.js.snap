// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="flatlist"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "yellow",
            "height": 20,
          }
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
